### ISSUE TITLE
FAST_LIVO2 collected data delta_timestamp is large

### ISSUE THREAD

**Issue by hmtccss** [hku-mars/M2Mapping#17]
Hello, very nice project. I collected data from FAST_LIVO2 , and the delta_timestamp is large, which larger than 0.01 ,that you set. So the parsed data is null.

**Comment by jianhengLiu** [comment]
It is an issue that FAST-LIVO2 outputs topics with ros::now() timestamps, which might cause a mismatch between pose and sensor data.  I am now occupied by an ongoing project, but I will definitely fix this issue in the future. If you urge it, I might guide you in fixing it and request a pull request to the modified FAST-LIVO2 codes for the community.

**Comment by hmtccss** [comment]
Can I manually align all the poses and sensor data at fixed intervals starting from the timestamp of the first frame? Would this affect the experimental results?

**Comment by jianhengLiu** [comment]
In FAST-LIVO2 datasets, it might be ok, as the datasets are collected in equipment with hardware time synchronization. Ideally, it outputs the same number of pointclouds and images. 
But in general, this operation is not recommended; the timeshift leads to pose misalignment, which will greatly degenerate the results.

**Comment by hmtccss** [comment]
I use the same number of pointclouds and images collected from FAST_LIVO2 on HK_campus to train M2Mapping, I can not get good performance, the PSNR is 19. But your dataset 's PSNR can up to 30. why? do you have any clue?

**Comment by hmtccss** [comment]
And also I use your provided data M2_HK_campus to train model, and I find the SDF acc is very low while the PSNR is 30. I set the SDF prediction accuracy to have a positive and negative error of ±1 cm.

**Comment by hmtccss** [comment]
Is the M2mapping sensitive with the data time synchronization？

**Comment by hmtccss** [comment]
Sorry, I have another question. Why the training dataset's PSNR in the last epoch of training process is different from the metric evaluation stage.

**Comment by jianhengLiu** [comment]
> I use the same number of pointclouds and images collected from FAST_LIVO2 on HK_campus to train M2Mapping, I can not get good performance, the PSNR is 19. But your dataset 's PSNR can up to 30. why? do you have any clue?

I suppose the sensor poses are misaligned. You can check the train views' color if they match the render depth

**Comment by jianhengLiu** [comment]
> And also I use your provided data M2_HK_campus to train model, and I find the SDF acc is very low while the PSNR is 30. I set the SDF prediction accuracy to have a positive and negative error of ±1 cm.

How do you set the prediction accuracy?
For outdoor datasets, +- 1cm might be too strict for SDF mapping.

**Comment by jianhengLiu** [comment]
> Is the M2mapping sensitive with the data time synchronization？

No quite relavent, but sensitive the pose accuracy.

**Comment by jianhengLiu** [comment]
> Sorry, I have another question. Why the training dataset's PSNR in the last epoch of training process is different from the metric evaluation stage.

One is evaluated in codes with float accuracy. The evaluation stage evaluates the output images after float->int, which might have a slightly lower PSNR.

**Comment by hmtccss** [comment]
> > Sorry, I have another question. Why the training dataset's PSNR in the last epoch of training process is different from the metric evaluation stage.
> 
> One is evaluated in codes with float accuracy. The evaluation stage evaluates the output images after float->int, which might have a slightly lower PSNR.

I train the collected data. The first training PSNR is 36, but the last evaluated PSNR is 22.  Amazing.

**Comment by hmtccss** [comment]
> > Sorry, I have another question. Why the training dataset's PSNR in the last epoch of training process is different from the metric evaluation stage.
> 
> One is evaluated in codes with float accuracy. The evaluation stage evaluates the output images after float->int, which might have a slightly lower PSNR.

I train the collected data. The  PSNR in training stage is 36, but the PSNR in evaluation stage is 22. Amazing.

**Comment by jianhengLiu** [comment]
> > > Sorry, I have another question. Why the training dataset's PSNR in the last epoch of training process is different from the metric evaluation stage.
> > 
> > 
> > One is evaluated in codes with float accuracy. The evaluation stage evaluates the output images after float->int, which might have a slightly lower PSNR.
> 
> I train the collected data. The PSNR in training stage is 36, but the PSNR in evaluation stage is 22. Amazing.

As the PSNR of the training stage is one training view evaluation, while the evaluation stage evaluates all test images

**Comment by jianhengLiu** [comment]
And have you checked that your intrinsic parameters are correct? The default params for the provided datasets are a 0.5 scale for fast evaluation, and the modified-FAST-LIVO2 outputs the original size images.

**Comment by hmtccss** [comment]
> And have you checked that your intrinsic parameters are correct? The default params for the provided datasets are a 0.5 scale for fast evaluation, and the modified-FAST-LIVO2 outputs the original size images.

I print the image size, which is render.size= torch.Size([1, 3, 512, 640]), gt.size= torch.Size([1, 3, 512, 640]). Is it right?

**Comment by hmtccss** [comment]
There is another question. Why your campus dataset get 799 images and 850+ depths after parser. I see your HK_Campus rosbag very alignment. How do you preprocess the rosbag.  Was the data well aligned during the collection process?

**Comment by jianhengLiu** [comment]
> > And have you checked that your intrinsic parameters are correct? The default params for the provided datasets are a 0.5 scale for fast evaluation, and the modified-FAST-LIVO2 outputs the original size images.
> 
> I print the image size, which is render.size= torch.Size([1, 3, 512, 640]), gt.size= torch.Size([1, 3, 512, 640]). Is it right?

It looks fine.

**Comment by jianhengLiu** [comment]
> There is another question. Why your campus dataset get 799 images and 850+ depths after parser. I see your HK_Campus rosbag very alignment. How do you preprocess the rosbag. Was the data well aligned during the collection process?

As we assume that the output depths and images are well aligned, while FAST-LIVO2 outputs sensor data and pose with ros::now() timestamps, which is not the actual capture time. It results in the misparsing or misaligned sensor number. And we are working on fixing the modified-FAST-LIVO2 to output sensor with capture timestamp

**Comment by jianhengLiu** [comment]
I have updated a version of FAST-LIVO2 to output data with capture timestamps. Please try if it fixes your issue
